composer/core/data_spec.py

```diff
@@ -246,6 +252,9 @@ def _default_get_num_samples_in_batch(self, batch: Batch) -> int:
         if isinstance(batch, torch.Tensor):
             return batch.shape[0]
 
+        if _TRANSFORMERS_INSTALLED and isinstance(batch, BatchEncoding):
```

Comment:
sorry i still don't get the need for this. `BatchEncoding` supports dict like access, and instead of checking `isinstance(batch, dict)` below, I think we just want to update to be `isinstance(batch, Mapping)`